pkg/controller/controller_utils.go

```diff
@@ -357,8 +440,33 @@ func ClusterAPIMachineProviderConfigFromMachineSetSpec(machineSetSpec *clusterop
 	}, nil
 }
 
-// PopulateMachineSpec ensures that the MachineSetSpec we use for the machine spec provider config is fully populated with defaults and calculated values based on Cluster Operator specific logic. This can be used both on individual machines, as well as on the MachineTemplateSpec used in MachineSets and MachineDeployments.
-func PopulateMachineSpec(machineSpec *clusterapi.MachineSpec, clusterSpec *clusteroperator.ClusterDeploymentSpec, clusterVersion *clusteroperator.ClusterVersion, mLog log.FieldLogger) error {
+// MachineProviderConfigFromMachineSetConfig returns a RawExtension with a machine ProviderConfig from a MachineSetConfig
+func MachineProviderConfigFromMachineSetConfig(machineSetConfig *clusteroperator.MachineSetConfig, clusterDeploymentSpec *clusteroperator.ClusterDeploymentSpec, clusterVersion *clusteroperator.ClusterVersion) (*runtime.RawExtension, error) {
+	msSpec := &clusteroperator.MachineSetSpec{
+		MachineSetConfig: *machineSetConfig,
+	}
+	vmImage, err := getImage(clusterDeploymentSpec, clusterVersion)
+	if err != nil {
+		return nil, err
+	}
+	msSpec.VMImage = *vmImage
+
+	// use cluster defaults for hardware spec if unset:
+	hwSpec, err := ApplyDefaultMachineSetHardwareSpec(msSpec.Hardware, clusterDeploymentSpec.DefaultHardwareSpec)
+	if err != nil {
+		return nil, err
+	}
+	msSpec.Hardware = hwSpec
+
+	// Copy cluster hardware onto the provider config as well. Needed when deleting a cluster in the actuator.
+	msSpec.ClusterHardware = clusterDeploymentSpec.Hardware
+
+	return ClusterAPIMachineProviderConfigFromMachineSetSpec(msSpec)
+}
+
+// PopulateMachineSpec ensures that the MachineSetSpec we use for the machine spec provider config is fully populated with defaults and calculated values based on Cluster Operator specific logic.
+// This can be used both on individual machines, as well as on the MachineTemplateSpec used in MachineSets and MachineDeployments.
+func PopulateMachineSpec(machineSpec *clusterapi.MachineSpec, clusterDeploymentSpec *clusteroperator.ClusterDeploymentSpec, clusterVersion *clusteroperator.ClusterVersion, mLog log.FieldLogger) error {
```

Comment:
We should delete this function. It isn't used. The Machine spec will not need to be populate from MachineSet and Cluster any more with the advent of ClusterDeployment. I meant to delete this with my PR but forgot.